### ISSUE TITLE
help_query.etz and other files are wrong for SVN and contain spelling errors 

### ISSUE THREAD

**Issue by warmsocks** [viewvc/viewvc#150]
I'm using ViewVC 1.1.26 on CentOS with Apache httpd 2.2.15 with an older Subversion server, version 1.6.11 (r934486).  Not sure what the version of Python is.  No CVS server.  Our server is behind a firewall and not publicly accessible but I've listed one below that is accessible that shows the same behavior.

We've turned on the Query Revision History feature, and I've noticed that the help file for the server  at viewvc-static/help_query.html contains the following errors:

### 1. Help text refers to CVS only
The help text begins with "Select your parameters for querying the CVS commit database", but we're not using CVS.  This is minor issue, but see item 3 below.

### 2. There are numerous spelling errors
The following words are all misspelled, sometimes repeatedly:

- _seperated_ should be _separated_
- _preceeds_ should be _precedes_
- _wildard_ should be _wildcard_
- _extentions_ should be _extensions_

This is distracting but also minor.

### 3. The wildcard and regex search documentation are out of date/just plain wrong for SVN
The docs for wildcard and regex read as follows:

> Wildcard and regular expression searches are entered in a similar way to the quoted requests. You must quote any wildcard or regular expression request, and a command character preceeds the first quote. The command character l(lowercase L) is for wildcard searches, and the wildcard character is a percent (%). The command character for regular expressions is r, and is passed directly to MySQL, so you'll need to refer to the MySQL manual for the exact regex syntax. It is very similar to Perl. A wildard search for all files with a .py extention is: l"%.py" in the File input box. The same search done with a regular expression is: r".*\.py".

I see that this applies to ViewVC 1.1.2 with CVS, as the searches work as described on this website:

http://cvs.opengroup.org/cgi-bin/query.cgi?repository=&branch=&directory=&file=+l%22%25.py%22&who=&sortby=date&hours=2&date=all

However, on the currently released ViewVC (1.1.26) with svn server 1.6.11, the glob and regex queries only work if the command character and quotes are _not_ present, and the wildcards used are the "normal" ones used in globs and regexes and not the % sign.

So for example a glob pattern match of `a*.cpp` works and and a regex of `^a.*\.cpp` works, but `l"a%.cpp"` and `r"^a.*\.cpp"` do not.

Although our server is not accessible, you can see the same behavior on this server, which is:

https://svnsrv.desy.de/viewvc/unfolding/RooUnfold/trunk/?view=query&dir=&file=.*%5C.icc%24&file_match=regex&who=&who_match=exact&comment=ROOT+6%5C.0&comment_match=regex&querysort=date&hours=2&date=all&mindate=&maxdate=&limit_changes=100

Copies of all the above errors are present in the following files on the 1.1.x branch.  It appears that although the files have been moved around on master, they still have the same issues.

```
templates-contrib/newvc/templates/docroot/help_query.html
templates-contrib/newvc/templates/query.ezt
templates-contrib/viewsvn/templates/query.ezt
templates/docroot/help_query.html
templates/include/paging.ezt
templates/include/sort.ezt
templates/query.ezt
```

I'm willing to fix this myself and submit a pull request for both the 1.1.x branch and master if I can get a few questions answered.  I've read docs/template-authoring-guide.html but the good part is still under development.  :)

**Comment by cmpilato** [comment]
Thanks for the issue report.  I'm happy to answer any questions you may have.

As you may know (or have discerned), the Subversion query support was a hack job just to get something working -- it uses the same DB schema as CVS does, which means it knows nothing of directories as versioned objects, knows nothing of copies and renames, etc.  This was a deliberate design decision at the time so that folks could start using the query with their CVS and Subversion repositories side-by-side and no DB migration to deal with.  That was yeeeeears ago.

I confess I'm a bit confused about the reported wildcard handling differences between CVS and Subversion, though, because there's no Subversion-specific code fork in the query logic.  _At all_.  One of the first things that code does is to throw all the configured Subversion repositories into the same `cvsroots` variable that the holds the configured CVS repositories, and from that point on the code is blissfully ignorant of which VCS was used to populate the database.  But then, I don't use the feature that much myself, so there's plenty I'm sure I don't understand.

As for the "good part" of `template-authoring-guide.html` being under development, I'm assuming you mean the TODO section under "Using EZT".  I never got around to fleshing that out because EZT has pretty decent self-contained docs.  Just check out the first 25% of https://github.com/viewvc/viewvc/blob/master/lib/ezt.py and I think you'll agree.

**Comment by warmsocks** [comment]
@cmpilato Ahhhh, thank you for the info and history.  I wasn't aware of the svn query support backstory.  That explains a few things -- but it is still good to have, even if it isn't perfect.  The other info helps too.  Thanks for being willing to answer Qs, and the pointer to the in-app ezt docs.  Very nice.  Putting a pointer in template-authoring-guide.html to them would be most excellent.

It's good to hear there is no svn-specific code fork, so the description can change without respect to whether the repo being queried is svn or cvs since search queries should be the same.  Something did change in the query logic somewhere between 1.1.2 and 1.1.26; if you look at the last two URLs in the original report, the 1.1.2 version uses query.cgi and embeds the match type --
 plain/wildcard/regex -- into the search term.  In contrast, 1.1.26 uses a REST-style API and specifies match type variables, and the query forms changed in matching ways between the two releases.  I just want to confirm that difference is due to a permanent across-the-board code change between 1.1.2 and 1.1.26, and the old `l"%.py"` style isn't still available when CVS repos are used, or available via a config option.  Certainly looks like the old way is gone for good, just double checking.

Is another 1.1.x release planned?  If so I'd like to submit a PR for 1.1.x first, since that is what we're running here and what I can test on, and a separate one for master after, since a lot of stuff under templates/ has been moved around, rearranged and modified for 1.2.  Does that work for you?

**Comment by cmpilato** [comment]
Here's the diff between the two release tags, if you're interested in exploring:  https://github.com/viewvc/viewvc/compare/1.1.2...1.1.26

But I think I see what's throwing you, and I apologize for not having noticed it earlier.

The comparison of the two URLs you included in your original comment is a bit of an apples to oranges one.  As you said, the one uses 1.1.2 and the `query.cgi` script and interface (which predates the 1.0 release of the product).  The other is using the integrated query view (`.../viewvc.cgi?view=query`) introduced in 1.0.0, which offers a different querying form altogether.  The old `query.cgi` approach is still available in 1.1.26 (for CVS repositories only, IIRC), and would -- I imagine -- behave the same as it did in 1.1.2.

It can be a bit confusing when looking at the templates and help stuff here, too.  `query.ezt` is the template for the old `query.cgi` script and approach, with `help_query.html` associated.  On the other hand, `query_form.ezt` and `query_results.ezt` drive the display of the newer integrated query interface, and don't appear to have a dedicated help file _at all_ as of this moment.

I do 1.1.x releases on occasion, as bugfixes accumulate.  I'd be happy to push out another such release with your doc improvements.

**Comment by cmpilato** [comment]
Well, I finally got around to fixing those spelling errors.  As for the overall short-comings of the online help for ViewVC, I've opened a new issue (#173) to track that situation.
